src/KeyringController.ts

```diff
@@ -273,26 +325,29 @@ class KeyringController extends EventEmitter {
    *
    * Only supports 'Simple Key Pair'.
    *
-   * @param {string} type - The key pair type to check for.
-   * @param {Array<string>} newAccountArray - Array of new accounts.
-   * @returns {Promise<Array<string>>} The account, if no duplicate is found.
+   * @param type - The key pair type to check for.
+   * @param newAccountArray - Array of new accounts.
+   * @returns The account, if no duplicate is found.
    */
-  async checkForDuplicate(type, newAccountArray) {
+  async checkForDuplicate(
+    type: string,
+    newAccountArray: string[],
+  ): Promise<string[]> {
     const accounts = await this.getAccounts();
 
     switch (type) {
-      case KEYRINGS_TYPE_MAP.SIMPLE_KEYRING: {
+      case KeyringType.Simple: {
         const isIncluded = Boolean(
           accounts.find(
-            (key) =>
+            (key: string) =>
               key === newAccountArray[0] ||
-              key === stripHexPrefix(newAccountArray[0]),
+              key === stripHexPrefix(newAccountArray[0] as string),
```

Comment:
```suggestion
              // strip hex prefix
              key === parseInt(newAccountArray[0] as string, 16).toString(16),
```

This would remove the need to declare the `HEX_PREFIX` constant, I think?

Reply:
[d0f6e63](https://github.com/MetaMask/KeyringController/pull/202/commits/d0f6e635771905e76c0e6f6d47eb874e8f28d16a)